### PR TITLE
Replace butler-auth with laravel sanctum

### PR DESCRIPTION
**Pull request by ttrig**
https://laravel.com/docs/8.x/sanctum

- [x] Wait for https://github.com/glesys/butler-service/pull/57
- [x] Test in production

### CODE REVIEW

**Review by christofferlindstrom — APPROVED** [review]

File: src/config/auth.php [hunk 1]
```diff
@@ -38,11 +38,12 @@
     'guards' => [
         'web' => [
             'driver' => 'session',
-            'provider' => 'users',
+            'provider' => 'consumers',
         ],
 
-        'api' => [
-            'driver' => 'jwt',
+        'sanctum' => [
```
Comment (ttrig) [inline src/config/auth.php:44]:
```suggestion
        'api' => [
```

File: src/routes.php [hunk 2]
```diff
@@ -17,7 +17,7 @@
     )->name('health');
 });
 
-Route::middleware(['api', 'auth'])->group(function () {
+Route::middleware(['api', 'auth:sanctum'])->group(function () {
```
Comment (ttrig) [inline src/routes.php:20]:
```suggestion
Route::middleware(['api', 'auth'])->group(function () {
```
Testa.

File: src/config/auth.php [hunk 3]
```diff
@@ -14,7 +14,7 @@
     */
 
     'defaults' => [
-        'guard' => env('AUTH_GUARD', 'api'),
+        'guard' => env('AUTH_GUARD', 'sanctum'),
```
Comment (ttrig) [inline src/config/auth.php:17]:
```suggestion
        'guard' => env('AUTH_GUARD', 'api'),
```

**Review by AvvyT — APPROVED** [review]

File: src/Testing/Concerns/MigratesDatabases.php [hunk 4]
```diff
@@ -20,11 +20,7 @@ public function migrateDatabase(string $database = 'default'): void
 
     private function migrationsPath(string $database): string
     {
-        if ($database === 'default' && ! is_dir(database_path('migrations/default'))) {
-            return 'database/migrations';
-        }
-
-        return "database/migrations/{$database}";
+        return $database === 'default' ? '' : "database/migrations/{$database}";
```
Comment (ttrig) [inline src/Testing/Concerns/MigratesDatabases.php:23]: `ServiceProvider::loadMigrations()` takes care of "default" migrations paths.

**Review by wecc — APPROVED** [review]

**Review by emil-nasso — APPROVED** [review]

**Review by olyckne — APPROVED** [review]

**Pull request merged**